### ISSUE TITLE
[isoltest] Add `balance:` statement

### ISSUE THREAD

**Issue by axic** [ethereum/solidity#10426]
Currently there is no simple way to query a contract balance in the semantic tests. Due to this lack of support, numerous tests were not moved from `SolidityEndToEndTest`.

(It is possible to add a helper in the contract, like `function balance() external view returns (uint) { return address(this).balance; }`, but that is a bit overkill.)

It would be nice to support a `// balance: 12 wei` or `// balance -> 12 wei` line to assert balance at that given point.

Could also consider checking the balance of any account, so perhaps the better syntax is:
```
// balance -> 12 wei   # for the current contract #
// balance 0x1234... -> 34 wei  # for address 0x1234... #
```

**Comment by chriseth** [comment]
Should be similar to `storage: empty / non-empty`

**Comment by axic** [comment]
> Should be similar to storage: empty / non-empty

That was the inspiration, but the second syntax proposal is better if we want to support querying any account (which I think would be useful to test transfers, etc.)

**Comment by chriseth** [comment]
the problem is that addresses are not fixed. If we want to query an external balance we could provide the contract name, but I think just supporting the balance of the "current" contract is already nice

**Comment by axic** [comment]
> the problem is that addresses are not fixed. 

I think they are, but agree if there is no such big need yet to check other balances, that can done via wrapper functions and we can change this as the need arises.

**Comment by axic** [comment]
The syntax implemented by #10873 is:
```
// balance -> 0
// balance: 0x0000000000000000000000000000000000000000 -> 0
```

The wei/ether suffix is not supported.
